kreuzberg/_extractors/_presentation.py

```diff
@@ -106,8 +124,49 @@ def _extract_pptx(self, file_contents: bytes) -> ExtractionResult:
             chunks=[],
         )
 
+        if self.config.extract_images:
+            result.images = self._extract_images_from_pptx(presentation)
+
         return self._apply_quality_processing(result)
 
+    def _extract_images_from_pptx(self, presentation: Presentation) -> list[ExtractedImage]:
```

Comment:
Code needs refactoring and optimizations - nested loops, try blocks, many ifs. Applies to entire file